include/drivers/gpio.h

```diff
@@ -30,14 +30,59 @@ extern "C" {
  * @brief GPIO Driver APIs
  * @defgroup gpio_interface GPIO Driver APIs
  * @ingroup io_interfaces
+ *
+ * @{
+ * @name GPIO Atomicity and Interrupt-Related Behavior
+ *
  * @{
+ * All GPIO peripheral API operations satisfy the following promises:
+ *
+ * * All functions shall be thread-safe and interrupt-safe.
+ * * All functions may be invoked from interrupt service routines and
+ *   callbacks.  Functions that cannot fulfill their behavioral
+ *   requirements in that context may return `-EWOULDBLOCK`.
+ * * All pin, interrupt, and callback configuration functions must be
+ *   atomic.
+ * * All functions that change output values must be atomic.  (For
```

Comment:
When we talk about thread/interrupt safety we say "all functions" without going into details which one. Since we want all functions to be atomic let's not mention categories, like pin, interrupt configuration but simply refer to "all functions".

Reply:
done